c++/src/kj/compat/tls.c++

```diff
@@ -445,8 +457,53 @@ public:
   }
 
 private:
+  Promise<void> onAcceptSuccess(AuthenticatedStream&& stream) {
+    auto acceptPromise = kj::evalNow([&] {
+      // Do the SSL acceptance procedure.
+      return tls.wrapServer(kj::mv(stream));
+    });
+
+    auto sslPromise = acceptPromise.then([this](auto&& stream) -> Promise<void> {
+      // This is only attached to the success path, thus the error handler will catch if our
+      // promise fails.
+      queue.push(kj::mv(stream));
+      return kj::READY_NOW;
+    });
+    tasks.add(kj::mv(sslPromise));
+
+    // Queue up the next accept loop immediately without waiting for SSL_accept()/wrapServer().
+    return acceptLoop();
+  }
+
+  void onAcceptFailure(Exception&& e) {
+    KJ_LOG(ERROR, "TlsConnectionReceiver could not accept()", e);
```

Comment:
Since we're propagating the exception to the app, I would leave it up to the app to decide whether to log.